src/reducers/scrollStateHelper.js

```diff
@@ -239,21 +248,18 @@ function calculateRenderedRowRange(state, scrollAnchor) {
 
   // Loop to walk the viewport until we've touched enough rows to fill its height
   let rowIdx = startIdx;
-  while (totalHeight < viewportHeight && rowIdx < rowsCount && rowIdx >= 0) {
+  let endIdx = rowIdx;
+  while (totalHeight < availableHeight && rowIdx < rowsCount && rowIdx >= 0) {
     totalHeight += updateRowHeight(state, rowIdx);
+    endIdx = rowIdx;
     rowIdx += step;
   }
 
-  const endIdx = rowIdx - step;
```

Comment:
Rewrote this to handle the case where no height is available better.  We don't want endIdx to ever be -1 (rowIdx=0 & step=1).